[MrAsparag00se] iP

MrAsparag00se opened the pull request:
# Vegetables: Task Management System 🥕 

Welcome to the Vegetables! 🥦 This bot is your new task management buddy, helping you organize your tasks with a simple yet effective interface.

### Supported Commands:

- **"help"** - Displays a list of available commands.
- **"list"** - Lists all tasks.
- **"todo [description]"** - Adds a new to-do task.
- **"deadline [description] /by [date]"** - Adds a new deadline task.
- **"event [description] /at [date]"** - Adds a new event task.
- **"mark [task number]"** - Marks a task as completed.
- **"unmark [task number]"** - Marks a task as incomplete.
- **"find [keyword]"** - Searches for tasks containing the given keyword.
- **"delete [task number]"** - Removes a task from the list.
- **"bye"** - Saves tasks and exits the application.

---

### How it Works:

1. **Add a task**: Use `"todo [description]"` to create new tasks.
2. **Set deadlines**: Use `"deadline [description] /by [date]"` to add deadlines.
3. **Events**: Use `"event [description] /at [date]"` to create events.
4. **Manage tasks**:
   - Mark tasks with `"mark [task number]"`.
   - Unmark tasks with `"unmark [task number]"`.
5. **Search tasks**: Use `"find [keyword]"` to search for tasks.
6. **Delete tasks**: Remove tasks with `"delete [task number]"`.
   
> _Tasks are stored persistently on your hard drive for easy access anytime!_

### Task Example:

```python
# Create a new task in Python
task = create_task("Finish Vegetables PR")
```
Todo List:
 - [x] Display available commands
 - [ ] Re-implement enums
 - [ ] Assertions
 - [ ] Checkstyle
 - [ ] J-Unit testing. URGENT!!

For more info, check out [Vegetables Chatbot Docs](https://github.com/MrAsparag00se/ip/blob/master/docs/README.md).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

MrAsparag00se committed:
Rename, Greet, Exit

MrAsparag00se committed:
Add echo functionality

MrAsparag00se committed:
Add message to list functionality

MrAsparag00se committed:
Add mark/unmark functionality

MrAsparag00se committed:
Add event,todo,deadline subclasses

MrAsparag00se committed:
Add basic error handling

MrAsparag00se committed:
Add delete functionality

MrAsparag00se committed:
Edit I/O redirection testing files

MrAsparag00se committed:
Add enum and refactor code

MrAsparag00se committed:
Add functionality to save tasks to hard disk

MrAsparag00se committed:
Resolve merge conflict

MrAsparag00se committed:
Process and filter dates and time for deadline

MrAsparag00se committed:
Add functionality for task searching

MrAsparag00se committed:
Maintain coding standard

MrAsparag00se committed:
Merge branch 'branch-A-CodingStandard'

MrAsparag00se committed:
Maintain coding standard again

MrAsparag00se committed:
Merge branch 'branch-A-CodingStandard'

Update code to reflect the coding standard

MrAsparag00se committed:
Add more OOP

MrAsparag00se committed:
Merge branch 'branch-A-MoreOOP'

Add more OOP

MrAsparag00se committed:
More-OOP here, without separate files for classes. Replacing missing branch-Level-8

MrAsparag00se committed:
Merge missing branch-Level-8 into master using More-OOP

MrAsparag00se committed:
Proper refactoring of classes, packaging, javadoc, code style

MrAsparag00se committed:
Merge remote-tracking branch 'origin/add-gradle-support'
Merge to add gradle support

MrAsparag00se committed:
Add Gradle

MrAsparag00se committed:
Add minimal JUnit tests

MrAsparag00se committed:
Partial solution for Level 10

MrAsparag00se committed:
Level-10 updates complete

MrAsparag00se committed:
C-BetterSearch

MrAsparag00se committed:
C-DetectDuplicates

MrAsparag00se committed:
B-DetectAnomalies: Invalid dates

MrAsparag00se committed:
B-DetectAnomalies: Overlapping events

MrAsparag00se committed:
Improve readability of TaskStorage class

MrAsparag00se committed:
Minor fix to close when bye command

MrAsparag00se committed:
Apply CheckStyle and rename VegetablesGUI

MrAsparag00se committed:
Merge branch-A-CheckStyle, no-ff

MrAsparag00se committed:
A-MoreEnum complete

MrAsparag00se committed:
Remove assertions for now, caused problems with gradlew test

MrAsparag00se committed:
Merge branch 'branch-A-MoreEnums'
Add proper enums for CommandHandler

MrAsparag00se committed:
Implement CheckStyle for test files

MrAsparag00se committed:
Add title to GUI window

MrAsparag00se committed:
Align date display format of Event with Deadline

MrAsparag00se committed:
Update README and add UI screenshot

MrAsparag00se committed:
Update Readme

MrAsparag00se committed:
Minor formatting change to Readme

MrAsparag00se committed:
Newline minor formatting change to Readme

josejhkim commented on src/main/java/vegetables/command/CommandHandler.java:
```diff
@@ -0,0 +1,291 @@
+ */
+public class CommandHandler {
+    private final TaskManager taskManager;
+    private final TaskStorage taskStorage;
```
Perhaps according to the coding standard, it might be better to rename your final fields using capital letters?
The same can be said for other 'final' fields in different files

josejhkim commented on src/main/java/vegetables/command/CommandHandler.java:
```diff
@@ -0,0 +1,291 @@
+            }
+            default -> "Unrecognised command!";
+        };
+    }
```
Perhaps it might be better for readability if you added line breaks after some of the function declarations for this file?

josejhkim commented on src/main/java/vegetables/command/CommandHandler.java:
```diff
@@ -0,0 +1,291 @@
+            return "Error: " + e.getMessage();
+        }
+    }
+}
```
I think adding JavaDocs even for your private methods may help, especially when debugging or if the codebase becomes very large

josejhkim reviewed (commented):
Looks great overall! I think fixing the (very) tiny nitpicks I mentioned may make it even better :)

MrAsparag00se committed:
Add some assertions

MrAsparag00se committed:
Add all assertions

MrAsparag00se committed:
Disable trailing whitespace check to allow flexible code formatting

The Checkstyle configuration previously enforced no trailing whitespace,
which caused formatting restrictions for certain chunks of code. For
readability and clarity, some sections of the code require spacing at
the end of lines.

To enable this flexibility and prevent unnecessary Checkstyle violations,
the trailing whitespace check has been disabled.

Let's:
* Remove the trailing whitespace check from the Checkstyle configuration.
* Allow for improved code readability, particularly when spacing is
  necessary between code chunks.

This change will make the code more readable and maintain the flexibility
to format code for clarity, especially when dealing with code blocks that
require more spacing.

MrAsparag00se committed:
Merge pull request #1 from MrAsparag00se/branch-A-Assertions

Add assertions to improve code correctness

MrAsparag00se committed:
Merge branch 'master' into branch-A-CodeQuality
Merge branch-A-CodeQuality with master

MrAsparag00se committed:
Merge pull request #2 from MrAsparag00se/branch-A-CodeQuality

Refactor code to improve readability and maintainability

MrAsparag00se committed:
Set up Continuous Integration (CI) with GitHub Actions for Gradle build

This commit adds a GitHub Actions workflow configuration file (gradle.yml) to automate the CI process for this project. The workflow ensures that Gradle builds and tests the project on multiple platforms (Ubuntu, macOS, and Windows) whenever a change is pushed or a pull request is made.

The CI setup includes:
- Setting up the necessary Java environment (JDK 17).
- Running Gradle tasks to validate the build and perform checks.
- Using GitHub's hosted runners for different operating systems to ensure cross-platform compatibility.

By automating this process, we can catch potential build errors early and ensure that changes work consistently across different environments.

This setup is important for maintaining code quality and reducing the likelihood of integration issues.

Tagging this commit as A-CI.

MrAsparag00se committed:
Fix indentation issues and remove suppressed warnings

- Resolved indentation problems related to 'case' and 'default' statements in the 'executeCommand' method.
- Removed @SuppressWarnings("CheckStyle") annotation after fixing the indentation to comply with the project’s code style.
- Ensured proper indentation for 'case' blocks in 'switch' statement according to Checkstyle rules.

MrAsparag00se committed:
Add better JavaDocs and coding standard

MrAsparag00se committed:
Add empty todo task error handling

MrAsparag00se committed:
Add delete with no argument error handling

MrAsparag00se committed:
Remove word art

MrAsparag00se committed:
Assist with AI to get fun fact generator

MrAsparag00se committed:
Add credit to AI for fun fact generator

MrAsparag00se committed:
Add more personality to responses

MrAsparag00se committed:
Add autoscroll functionality to GUI

MrAsparag00se committed:
Scale textbox with window

MrAsparag00se committed:
Fix bug not displaying full list in GUI

MrAsparag00se committed:
Fix bug unable to add second event

MrAsparag00se committed:
Format JavaDocs correctly

MrAsparag00se committed:
Update README.md

MrAsparag00se committed:
Update JAR for Linux
